### ISSUE TITLE
unnecessary_override reports method M when super method is @protected and M is not

### ISSUE THREAD

**Issue by srawlins** [dart-lang/linter#3475]
**Describe the issue**
A method override might exist to make a `@protected` method public, which does not seem to violate the existing [protected contract](https://pub.dev/documentation/meta/latest/meta/protected-constant.html). Such methods should not be reported as unnecessary. I feel like they're similar to an override which changes the types of parameters.

**To Reproduce**

```dart
// lib1.dart

class C {
  @protected
  void m() {}
}

class D extends C {
  @override
  void m() => super.m();
}

// lib2.dart

import 'lib1.dart';

void f() => D().m();
```

**Expected behavior**

No `unnecessary_overrides` report.
